galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java

```diff
@@ -0,0 +1,120 @@
+        if (errorsList.size() >0){
+            response = getResponseBuilder().buildResponse(response, "application/json", gson.toJson(errorsList), HttpServletResponse.SC_BAD_REQUEST);
+        }else{
+            response = getResponseBuilder().buildResponse(response, "application/json", "", HttpServletResponse.SC_OK);
+        }
+        return response;
+
+    }
+
+    public List<String> processRequest(String jsonBody) throws InternalServletException{
+        
+        JsonObject body = gson.fromJson(jsonBody, JsonObject.class);
+        String action = body.get("action").getAsString().toLowerCase().trim();
+        if (validActions.contains(action)){
+            JsonArray jsonArray = body.get("data").getAsJsonArray();
+            processDataArray(jsonArray, action);
+        }else{
+            ServletError error = new ServletError(GAL5025_UNSUPPORTED_ACTION, action);
+            throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
+        }
+    return errors;
+    }
+
+    public void processDataArray(JsonArray jsonArray, String action) throws InternalServletException{
+        try{
+            for (JsonElement element: jsonArray){
+                JsonObject resource = element.getAsJsonObject();
+                String kind = resource.get("kind").getAsString();
+                switch (kind){
+                    case "GalasaProperty":
+                        processGalasaProperty(resource,action);
+                        break;
+                    default:
+                        ServletError error = new ServletError(GAL5026_UNSUPPORTED_RESOURCE_TYPE,kind);
+                        throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
+                }
+            }
+        }catch(InternalServletException s){
+            errors.add(s.getMessage());
+        }catch(Exception e){
+            ServletError error = new ServletError(GAL5000_GENERIC_API_ERROR);
+            throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+        }
+    }
+    
+    public void processGalasaProperty (JsonObject resource, String action) throws InternalServletException{
+        String apiversion = resource.get("apiVersion").getAsString();
+        if (apiversion.equals(new GalasaProperty(null,null, null).getApiVersion())){
+            try{
+                GalasaProperty property = gson.fromJson(resource, GalasaProperty.class);
+                if (propertyUtility.isPropertyValid(property)){
+                    propertyUtility.setGalasaProperty(property, action);
+                }
+            }catch (InternalServletException i){
+                throw i;
+            }catch (Exception e){
+            ServletError error = new ServletError(GAL5000_GENERIC_API_ERROR, e.getMessage());
+            throw new InternalServletException(error, HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+            }
```

Comment:
Not sure if we need this try-catch block since we have a matching try-catch block in `processDataArray` which calls this method